cypress/e2e/finance/fiscalYears/calulate-available-balance-for-negative-amount.cy.js

```diff
@@ -0,0 +1,159 @@
+        ledgerId: ledger.id,
+      },
+    };
+    const budgets = {
+      first: {
+        ...Budgets.getDefaultBudget(),
+        allocated: 100,
+        fiscalYearId: fiscalYear.id,
+        fundId: funds.first.id,
+      },
+      second: {
+        ...Budgets.getDefaultBudget(),
+        allocated: 0,
+        fiscalYearId: fiscalYear.id,
+        fundId: funds.second.id,
+      },
+    };
+    const testData = {
+      organization: NewOrganization.getDefaultOrganization(),
+      fiscalYear,
+      user: {},
+    };
+
+    before('Create test data', () => {
+      cy.getAdminToken().then(() => {
+        FiscalYears.createViaApi(fiscalYear);
+        Ledgers.createViaApi(ledger);
+        Object.values(funds).forEach((fund) => {
+          Funds.createViaApi(fund);
+        });
+        Object.values(budgets).forEach((budget) => {
+          Budgets.createViaApi(budget);
+        });
+
+        Organizations.createOrganizationViaApi(testData.organization).then(() => {
+          testData.order = NewOrder.getDefaultOrder({ vendorId: testData.organization.id });
+          testData.orderLine = BasicOrderLine.getDefaultOrderLine({
+            listUnitPrice: 100,
+            fundDistribution: [{ code: funds.second.code, fundId: funds.second.id, value: 100 }],
+          });
+
+          Orders.createOrderWithOrderLineViaApi(testData.order, testData.orderLine).then(
+            (order) => {
+              testData.order = order;
+
+              OrderLines.getOrderLineViaApi({ query: `poLineNumber=="*${order.poNumber}*"` }).then(
+                (orderLines) => {
+                  testData.orderLine = orderLines[0];
+
+                  Invoices.createInvoiceWithInvoiceLineViaApi({
+                    vendorId: testData.organization.id,
+                    fiscalYearId: testData.fiscalYear.id,
+                    poLineId: testData.orderLine.id,
+                    fundDistributions: testData.orderLine.fundDistribution,
+                    accountingCode: testData.organization.erpCode,
+                    subTotal: 10,
+                  }).then((invoice) => {
+                    testData.invoice = invoice;
+
+                    Invoices.changeInvoiceStatusViaApi({
+                      invoice: testData.invoice,
+                      status: INVOICE_STATUSES.PAID,
+                    });
+                  });
+                },
+              );
+            },
+          );
+        });
+      });
+
+      cy.createTempUser([
+        Permissions.uiFinanceViewFiscalYear.gui,
+        Permissions.uiFinanceViewFundAndBudget.gui,
+      ]).then((userProperties) => {
+        testData.user = userProperties;
+
+        cy.login(testData.user.username, testData.user.password, {
+          path: TopMenu.fiscalYearPath,
+          waiter: FiscalYears.waitLoading,
+        });
+      });
+    });
+
+    after('Delete test data', () => {
+      cy.getAdminToken().then(() => {
+        Users.deleteViaApi(testData.user.userId);
```

Comment:
Please also delete other test datas, like: `Organizations`, `Orders`, `Invoices` ...

Reply:
When Invoice is turned to Paid status system creates transaction. If there are any transaction created it is not possible to delete related objects.